tensorpipe/CMakeLists.txt

```diff
@@ -206,29 +202,56 @@ if(TP_ENABLE_IBV)
     transport/ibv/reactor.cc
     transport/ibv/sockaddr.cc
     transport/ibv/utility.cc)
-  set(TP_PUBLIC_HEADERS ${TP_PUBLIC_HEADERS}
+  list(APPEND TP_PUBLIC_HDRS
     transport/ibv/error.h
     transport/ibv/factory.h
     transport/ibv/utility.h)
   set(TENSORPIPE_HAS_IBV_TRANSPORT 1)
 endif()
 
-if(APPLE)
-  find_library(CF CoreFoundation)
-  find_library(IOKIT IOKit)
-  target_link_libraries(tensorpipe PRIVATE ${CF} ${IOKIT})
+# Handling the file lists
+if(TP_USE_CUDA)
+  list(APPEND TP_PRIVATE_SRCS ${TP_CUDA_PRIVATE_SRCS})
+  list(APPEND TP_PUBLIC_HDRS ${TP_CUDA_HDRS})
 endif()
 
+# Add the library target
+add_library(tensorpipe ${TP_STATIC_OR_SHARED} ${TP_PUBLIC_SRCS})
+target_sources(tensorpipe PRIVATE ${TP_PRIVATE_SRCS})
```

Comment:
As mentioned above I'm convinced these two can be joined. The former distinction was probably some cruft.

Reply:
done.